core/src/main/scala/org/apache/spark/rdd/AsyncRDDActions.scala

```diff
@@ -123,3 +127,8 @@ class AsyncRDDActions[T: ClassTag](self: RDD[T]) extends Serializable with Loggi
       (index, data) => Unit, Unit)
   }
 }
+
+private object AsyncRDDActions {
+  val futureExecutionContext = ExecutionContext.fromExecutorService(
+    ThreadUtils.newDaemonCachedThreadPool("AsyncRDDActions-future", 1024))
```

Comment:
is 1024 too high? In reality, I wonder if we are going to hit this high number of threads.